### PR TITLE
Update autorest.csharp and fix additional properties API usages

### PR DESCRIPTION
**Pull request by chamons**
In fixing https://github.com/Azure/autorest.csharp/pull/1284 for new SDKs, we've removed the IDictionary interface from new models that contain an AdditionalProperties property and made it public by default.

To not cause API breaking changes, this PR removes those changes by adding customizations to the affected SDKs.

Except Synapse, which I personally own, has not gone GA, and I'm signing off as fine w \ the change.

# All SDK Contribution checklist:

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] **Please open PR in `Draft` mode if it is:**
	- Work in progress or not intended to be merged.
	- Encountering multiple pipeline failures and working on fixes.
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/dotnet/corefx/blob/master/Documentation/coding-guidelines/breaking-change-rules.md).**

### [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/master/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code. (Track 2 only)
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK. Please double check nuget.org current release version.

## Additional management plane SDK specific contribution checklist: 
Note: Only applies to `Microsoft.Azure.Management.[RP]` or `Azure.ResourceManager.[RP]`
 
- [ ] Include updated [management metadata](https://github.com/Azure/azure-sdk-for-net/tree/master/eng/mgmt/mgmtmetadata).
- [ ] Update AzureRP.props to add/remove version info to maintain up to date API versions.

### Management plane SDK Troubleshooting
- If this is very first SDK for a services and you are adding new service folders directly under /SDK, please add `new service` label and/or contact assigned reviewer.
- If the check fails at the `Verify Code Generation` step, please ensure:
	- Do not modify any code in generated folders.
	- Do not selectively include/remove generated files in the PR.
	- Do use `generate.ps1/cmd` to generate this PR instead of calling `autorest` directly.
	Please pay attention to the @microsoft.csharp version output after running `generate.ps1`. If it is lower than current released version (2.3.82), please run it again as it should pull down the latest version.
	
	**Note: We have recently updated the PSH module called by `generate.ps1` to emit additional data. This would help reduce/eliminate the Code Verification check error. Please run following command**:

	    `dotnet msbuild eng/mgmt.proj /t:Util /p:UtilityName=InstallPsModules`

### Old outstanding PR cleanup
 Please note:
	If PRs (including draft) has been out for more than 60 days and there are no responses from our query or followups, they will be closed to maintain a concise list for our reviewers.

### CODE REVIEW

**Comment by chamons** [comment]
All of synapse we're accepting since it's not GA.

**Comment by chamons** [comment]
Because this change has a _huge_ synapse API change, the diff view may not be useful due to the number of generated files.

Here is the diff (from tldr diff) of all non-generated files changed: https://gist.github.com/79a83bbc752f92ae6928a8092b3d2210

And here is a much smaller one excluding synapse: https://gist.github.com/chamons/4c61ba7dd4a551d39dccb8aa6cc1d714

File: sdk/search/Azure.Search.Documents/src/Models/AnswerResult.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+#nullable disable
+
+using System.Collections;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.Search.Documents.Models
+{
+    public partial class AnswerResult : IReadOnlyDictionary<string, object>
```
Comment (Mohit-Chakraborty) [inline sdk/search/Azure.Search.Documents/src/Models/AnswerResult.cs:14]: Why is this change needed?

Comment (chamons) [reply sdk/search/Azure.Search.Documents/src/Models/AnswerResult.cs:14]: Apologies for not writing a description in the PR before slapping it out of draft. Check now, and let me know if you have questions.

If Azure.Search.Documents has not gone GA yet, and you'd like to drop the IDictionary interface and just have AdditionalProperties public let me know and I'll revert that bit. We are assuming by default no API change unless asked.

Comment (Mohit-Chakraborty) [reply sdk/search/Azure.Search.Documents/src/Models/AnswerResult.cs:14]: Thanks, got it. We can remove the `IDictionary` interface from `AnswerResult` and `CaptionResult` as they are not part of the [latest GA](https://github.com/Azure/azure-sdk-for-net/blob/Azure.Search.Documents_11.3.0/sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs).

It should stay on `FacetResult`. However, shouldn't then there be no change in the one generated file? Or, if we are splitting it up, the new file can also be under Generated\Models\ (something like FacetResult.Dictionary.cs).

Comment (chamons) [reply sdk/search/Azure.Search.Documents/src/Models/AnswerResult.cs]: Removed in those two as requested - https://github.com/Azure/azure-sdk-for-net/pull/21778/commits/2efb27f00e53716ca61091aaa9c23d2def789129

On FacetResult we add the elements no longer code gen'd [here](https://github.com/Azure/azure-sdk-for-net/blob/b8d45e2478f1404bc22823dbf60885c8b43ea36a/sdk/search/Azure.Search.Documents/src/Models/FacetResult.cs#L10) and remove them from [here](https://github.com/Azure/azure-sdk-for-net/blob/b8d45e2478f1404bc22823dbf60885c8b43ea36a/sdk/search/Azure.Search.Documents/src/Generated/Models/FacetResult.cs).

Comment (Mohit-Chakraborty) [reply sdk/search/Azure.Search.Documents/src/Models/AnswerResult.cs]: > However, shouldn't then there be no change in the one generated file? Or, if we are splitting it up, the new file can also be under Generated\Models\ (something like FacetResult.Dictionary.cs).

Ignore this comment. I see now what you are doing and how the file is being reused for the purpose of implementing `IReadOnlyDictionary`.

**Review by Mohit-Chakraborty — APPROVED** [review]

File: sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs [hunk 2]
```diff
@@ -2321,21 +2321,14 @@ public partial class WordDelimiterTokenFilter : Azure.Search.Documents.Indexes.M
 }
 namespace Azure.Search.Documents.Models
 {
-    public partial class AnswerResult : System.Collections.Generic.IEnumerable<System.Collections.Generic.KeyValuePair<string, object>>, System.Collections.Generic.IReadOnlyCollection<System.Collections.Generic.KeyValuePair<string, object>>, System.Collections.Generic.IReadOnlyDictionary<string, object>, System.Collections.IEnumerable
+    public partial class AnswerResult
```
Comment (pakrym) [inline sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:2324]: We should fix this.

Comment (pakrym) [reply sdk/search/Azure.Search.Documents/api/Azure.Search.Documents.netstandard2.0.cs:2324]: Re-read comments.

**Review by pakrym — APPROVED** [review]
